include/translate-option.php

```diff
@@ -395,17 +410,27 @@ protected function check_value_recursive( $old_values, $values, $key, $mo ) {
 	}
 
 	/**
-	 * Sanitizes the option value.
+	 * Sanitizes the string translation.
 	 *
 	 * @since 2.9
+	 * @since 3.7 Add $context and $original parameters.
 	 *
-	 * @param string $value The unsanitised value.
-	 * @param string $name  The name of the option.
+	 * @param string $value    The unsanitised string translation value.
+	 * @param string $name     The name registered for the string.
+	 * @param string $context  The context registered for the string.
+	 * @param string $original The original string to translate.
 	 * @return string Sanitized value.
 	 */
-	public function sanitize_option( $value, $name ) {
-		/** @var string $value */
-		$value = sanitize_option( $name, $value );
-		return $value;
+	public function sanitize_option( $value, $name, $context, $original ) {
+		if ( ! in_array( md5( "$original|$name|$context" ), $this->hashes, true ) ) {
+			return $value;
+		}
+
+		if ( is_callable( $this->sanitize_callback ) ) {
+			return call_user_func( $this->sanitize_callback, $value, $name, $context, $original );
+		}
+
+		/** @var string */
+		return sanitize_option( $name, $value );
```

Comment:
```suggestion
		return call_user_func( $this->sanitize_callback, $value, $name, $context, $original );
```
Follows previous suggestions.